src/dotnet-roslyn-tools/PRTagger/PRTagger.cs

```diff
@@ -2,133 +2,256 @@
+        if (currentBuild is null)
+        {
+            logger.LogError($"{gitHubRepoName} build not found for VS commit SHA {currentBuild}.");
+            return TagResult.Failed;
+        }
+
+        if (previousBuild is null)
+        {
+            logger.LogError($"{gitHubRepoName} build not found for VS commit SHA {previousBuild}.");
+            return TagResult.Failed;
+        }
+
+        // If builds are the same, there are no PRs to tag
+        if (currentBuild.Equals(previousBuild))
+        {
+            logger.LogInformation($"No PRs found to tag; {gitHubRepoName} build numbers are equal: {currentBuild}.");
+            return TagResult.NoChangeBetweenVSBuilds;
+        }
+
+        // Get commit SHAs for product builds
+        var previousProductCommitSha = await TryGetProductCommitShaFromBuildAsync(product, connections, previousBuild, logger).ConfigureAwait(false);
+        var currentProductCommitSha = await TryGetProductCommitShaFromBuildAsync(product, connections, currentBuild, logger).ConfigureAwait(false);
+
+        if (previousProductCommitSha is null || currentProductCommitSha is null)
+        {
+            logger.LogError($"Error retrieving {gitHubRepoName} commit SHAs.");
+            return TagResult.Failed;
+        }
+
+        logger.LogInformation($"Finding PRs between {gitHubRepoName} commit SHAs {previousProductCommitSha} and {currentProductCommitSha}.");
+
+        // Retrieve GitHub repo
+        string? gitHubRepoPath;
+        try
+        {
+            gitHubRepoPath = Environment.CurrentDirectory + "\\" + gitHubRepoName;
+            if (!Repository.IsValid(gitHubRepoPath))
+            {
+                logger.LogInformation("Cloning GitHub repo...");
+                gitHubRepoPath = Repository.Clone(product.RepoHttpBaseUrl, workdirPath: gitHubRepoPath);
+            }
+            else
             {
-                logger.LogError($"Error retrieving {gitHubRepoName} commit SHAs.");
-                continue;
+                logger.LogInformation($"Repo already exists at {gitHubRepoPath}");
             }
+        }
+        catch (Exception ex)
+        {
+            logger.LogError($"Exception while cloning repo: " + ex);
+            return TagResult.Failed;
+        }
+
+        // Find PRs between product commit SHAs
+        var prDescription = new StringBuilder();
+        var isSuccess = PRFinder.PRFinder.FindPRs(previousProductCommitSha, currentProductCommitSha, PRFinder.PRFinder.DefaultFormat, logger, gitHubRepoPath, prDescription);
+        if (isSuccess != 0)
+        {
+            // Error occurred; should be logged in FindPRs method
+            return TagResult.Failed;
+        }
+
+        var issueTitle = $"[Automated] PRs inserted in VS build {vsBuild}";
+        var hasIssueAlreadyCreated = await HasIssueAlreadyCreatedAsync(remoteConnections.GitHubClient, gitHubRepoName, issueTitle, logger).ConfigureAwait(false);
```

Comment:
Should this check happen earlier to avoid making any requests that aren't necessary?